app/channel.go

```diff
@@ -144,10 +144,6 @@ func (a *App) CreateChannelWithUser(channel *model.Channel, userId string) (*mod
 		return nil, model.NewAppError("CreateChannelWithUser", "api.channel.create_channel.direct_channel.app_error", nil, "", http.StatusBadRequest)
 	}
 
-	if strings.Index(channel.Name, "__") > 0 {
```

Comment:
Same here, I would expect to have an `IsValid` here, or may one in the `CreateChannel` method, this way we have the creation and update properly verified.